mypyc/lib-rt/int_ops.c

```diff
@@ -581,3 +581,59 @@ double CPyTagged_TrueDivide(CPyTagged x, CPyTagged y) {
     }
     return 1.0;
 }
+
+// int.to_bytes(length, byteorder, signed=False)
+PyObject *CPyTagged_ToBytes(CPyTagged self, Py_ssize_t length, PyObject *byteorder, int signed_flag) {
+    PyObject *pyint = CPyTagged_StealAsObject(self);
+    if (!PyLong_Check(pyint)) {
+        Py_DECREF(pyint);
+        PyErr_SetString(PyExc_TypeError, "self must be int");
+        return NULL;
+    }
+    if (!PyUnicode_Check(byteorder)) {
+        Py_DECREF(pyint);
+        PyErr_SetString(PyExc_TypeError, "byteorder must be str");
+        return NULL;
+    }
+    const char *order = PyUnicode_AsUTF8(byteorder);
+    if (!order) {
+        Py_DECREF(pyint);
+        return NULL;
+    }
+    PyObject *result = CPyLong_ToBytes(pyint, length, order, signed_flag);
+    Py_DECREF(pyint);
+    return result;
+}
+
+
+// Helper for CPyLong_ToBytes (Python 3.2+)
+PyObject *CPyLong_ToBytes(PyObject *v, Py_ssize_t length, const char *byteorder, int signed_flag) {
+    // This is a wrapper for PyLong_AsByteArray and PyBytes_FromStringAndSize
+    unsigned char *bytes = (unsigned char *)PyMem_Malloc(length);
```

Comment:
I think you can avoid allocating a temporary buffer if you call `PyBytes_FromStringAndSize` with `NULL` as the first argument to make the object uninitialized. You can then pass a pointer to the contents of the `bytes` object to `_PyLong_AsByteArray`.